### PR TITLE
Add a ENV tests with special chars in the values

### PR DESCRIPTION
**Pull request by duglin**
Tests a little bit of escaping quotes too

See https://github.com/docker/docker/pull/10431#issuecomment-75163177

Signed-off-by: Doug Davis dug@us.ibm.com

### CODE REVIEW

File: integration-cli/docker_cli_build_test.go [hunk 1]
```diff
@@ -3753,6 +3753,22 @@ ENV    abc 'yyy'
 RUN    [ $abc = \'yyy\' ]
 ENV    abc=
 RUN    [ "$abc" = "" ]
+ENV    abc=\$foo
```
Comment (thockin) [inline integration-cli/docker_cli_build_test.go]: I would argue for a testcase for 

ENV abc=$foo
ENV abc $foo

which seems to gibe you full coverage of escaped, unescaped, both styles, $, ', and "

Comment (duglin) [inline integration-cli/docker_cli_build_test.go]: I think we have the 2nd one covered here:
https://github.com/docker/docker/blob/master/integration-cli/docker_cli_build_test.go#L3706
but I don't see the 1st - will add it.

**Comment by LK4D4** [comment]
LGTM

File: integration-cli/docker_cli_build_test.go [hunk 2]
```diff
@@ -3753,6 +3756,22 @@ ENV    abc 'yyy'
...
+RUN    [ "$abc" = "\$foo" ]
```
Comment (tiborvass) [inline integration-cli/docker_cli_build_test.go]: @duglin Could be dangerous, if the builder substitutes both `\$foo` in RUN and in ENV. Then you have double negative yielding a positive result :(

Comment (duglin) [inline integration-cli/docker_cli_build_test.go]: @tiborvass  good catch. I've updated it - see what you think

**Comment by jfrazelle** [comment]
LGTM

**Pull request merged**